pkg/webhooks/machine_webhook.go

```diff
@@ -859,6 +865,21 @@ func validateAzure(m *machinev1.Machine, config *admissionConfig) (bool, []strin
 		errs = append(errs, field.Invalid(field.NewPath("providerSpec", "osDisk", "diskSizeGB"), providerSpec.OSDisk.DiskSizeGB, "diskSizeGB must be greater than zero and less than 32768"))
 	}
 
+	if providerSpec.OSDisk.DiskSettings.EphemeralStorageLocation != azureEphemeralStorageLocationLocal && providerSpec.OSDisk.DiskSettings.EphemeralStorageLocation != "" {
+		errs = append(errs, field.Invalid(field.NewPath("providerSpec", "osDisk", "diskSettings", "ephemeralStorageLocation"), providerSpec.OSDisk.DiskSettings.EphemeralStorageLocation,
+			fmt.Sprintf("osDisk.diskSettings.ephemeralStorageLocation can be either empty or set to %s", azureEphemeralStorageLocationLocal)))
+	}
+
+	cachingTypes := []string{azureCachingTypeNone, azureCachingTypeReadOnly, azureCachingTypeReadWrite}
+	if !stringInSlice(providerSpec.OSDisk.CachingType, cachingTypes) && providerSpec.OSDisk.CachingType != "" {
+		errs = append(errs, field.Invalid(field.NewPath("providerSpec", "osDisk", "cachingType"), providerSpec.OSDisk.CachingType,
+			fmt.Sprintf("osDisk.cachingType can be only %s or an empty string", strings.Join(cachingTypes, ", "))))
+	}
```

Comment:
Alternatively you can use a switch here, and you can avoid the list iteration `stringInSlice`
```suggestion
	switch providerSpec.OSDisk.CachingType:
	case azureCachingTypeNone, azureCachingTypeReadOnly, azureCachingTypeReadWrite, "":
	  // Valid scenarios, do nothing
	default:
	  errs = append(errs, field.Invalid(field.NewPath("providerSpec", "osDisk", "cachingType"), providerSpec.OSDisk.CachingType,
			fmt.Sprintf("osDisk.cachingType can be only %s, %s, %s or omitted", azureCachingTypeNone, azureCachingTypeReadOnly, azureCachingTypeReadWrite)))	
	}
```